fuel-core/src/database.rs

```diff
@@ -248,3 +260,132 @@ impl InterpreterStorage for Database {
         Ok(block.headers.producer)
     }
 }
+
+#[async_trait]
+impl RelayerDb for Database {
+    async fn get_validators(&self) -> HashMap<Address, u64> {
+        struct WrapAddress(pub Address);
+        impl From<Vec<u8>> for WrapAddress {
+            fn from(i: Vec<u8>) -> Self {
+                Self(Address::try_from(i.as_ref()).unwrap())
+            }
+        }
+        let mut out = HashMap::new();
+        for diff in self.iter_all::<WrapAddress, u64>(columns::VALIDATOR_SET, None, None, None) {
+            match diff {
+                Ok((address, stake)) => {
+                    out.insert(address.0, stake);
+                }
+                Err(_) => return out,
+            }
+        }
+        out
+    }
+
+    async fn get_validator_diffs(
+        &self,
+        from_fuel_block: u64,
+        to_fuel_block: Option<u64>,
+    ) -> Vec<(u64, HashMap<Address, u64>)> {
+        let to_fuel_block = if let Some(to_fuel_block) = to_fuel_block {
+            if from_fuel_block > to_fuel_block {
+                return Vec::new();
+            }
+            to_fuel_block
+        } else {
+            u64::MAX
+        };
+        struct WrapU64Be(pub u64);
```

Comment:
If we used a new-type for the da_height, we could just impl Into/TryFrom Vec<u8> and avoid all these inline structs.

Reply:
Problem is that it requires `From` and not `TryFrom` so i will still need wrapper type. Maybe changing `iter_all` to take `TryFrom` is a way to do it

Reply:
I have just read this again, new type seems like a way to go :)